core/arch/arm/plat-k3/drivers/mailbox.c

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Texas Instruments K3 Mailbox Driver
+ * Copyright (C) 2025 Texas Instruments Incorporated - https://www.ti.com/
+ */
+
+#include <io.h>
+#include <kernel/delay.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "ti_sci_protocol.h"
+#include "ti_sci_transport.h"
+
+/**
+ * TI K3 Mailbox structure
+ * @tx_base: Base address for the transmit mailbox
+ * @rx_base: Base address for the receive mailbox
+ * @tx_sram_va: Virtual address for the transmit mailbox SRAM
+ */
+struct ti_k3_mailbox_cfg {
+	vaddr_t tx_base;
+	vaddr_t rx_base;
+	vaddr_t tx_sram_va;
+} mailbox_obj;
+
+static struct ti_k3_mailbox_cfg *mailbox_ctx = &mailbox_obj;
```

Comment:
No need for this line. In fact there is no need for these to be inside a structure, there can only be one instance of the mailbox so just drop the structure and just access the TX/RX base addresses directly.